### PR TITLE
Feature: Add combined sensor sessions

### PR DESCRIPTION
**Pull request by time-trader**
# Contents ([#19](https://github.com/aerosense-ai/aerosense-tools/pull/19))

### Enhancements
- Compute sensor statistics
- Merge Session dataframes
- Raise Exceptions for empty dataframes
- Avoid extrapolation during merge and resampling to new time vector

### Dependencies
- Remove scipy dependency

### Testing
- Add pre-process tests

### CODE REVIEW

File: tests/test_pre_process.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from aerosense_tools.preprocess import RawSignal
+from aerosense_tools.preprocess import SensorMeasurementSession
+
+REPOSITORY_ROOT = os.path.dirname(os.path.dirname(__file__))
+TEST_START_TIME = dt.datetime(2020, 5, 4)
+TEST_DURATION = dt.timedelta(seconds=1)
+TEST_SAMPLING_STEP = dt.timedelta(seconds=0.1)
+
+
+class TestPreProcess(unittest.TestCase):
```
Comment (cortadocodes) [inline tests/test_pre_process.py:17]: Could you add docstrings to each method and each test?

File: tests/test_pre_process.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        merged_df = measurement_sessions1[0].merge_with_and_interpolate(measurement_sessions2[0])
+        self.assertTrue(pd.isna(merged_df.iloc[-1, -1]))
+        self.assertAlmostEqual((data2.iloc[0,0]+data2.iloc[1,0])/2, merged_df.iloc[0,2], 12)
+
+
+
+
+
+
```
Comment (cortadocodes) [inline tests/test_pre_process.py:83]: Excess whitespace

**Review by cortadocodes — CHANGES_REQUESTED** [review]
Looks good to me apart from missing docstrings

**Comment by time-trader** [comment]
> Looks good to me apart from missing docstrings

Done. Thank you for the review.

**Pull request merged**